syntax = "proto3";
package uranium.admin;
import "common.proto";

service UraniumAdminService {
  rpc CreateTable(TableOptions) returns (common.Result) {}
  rpc UpdateTable(TableOptions) returns (common.Result) {}
  rpc DropTable(common.TableName) returns (common.Result) {}
  rpc GetTableOptions(common.TableName) returns(TableOptions) {}
}

message TableOptions {
  common.TableName table_name = 1;
  oneof options {
    SchemaTableOptions schema_table_options = 2;
    SchemalessTableOptions schemaless_table_options = 3;
  }
}

enum StorageType {
  FLAT = 0;
  COLUMN_SEPARATED = 1;
  COLUMN_GROUP = 2;
}

message SchemaTableOptions {
  StorageType type = 1;
  repeated common.ColumnDefination columns = 3;
}

message SchemalessTableOptions {
  StorageType type = 1;
}

